cpp/src/search/contains_nested.cu

```diff
@@ -49,50 +49,39 @@ bool contains_nested_element(column_view const& haystack,
   auto const end   = begin + haystack.size();
   using cudf::experimental::row::rhs_index_type;
 
-  auto const found_it = [&] {
-    if (haystack.has_nulls()) {
-      auto const haystack_cdv_ptr  = column_device_view::create(haystack, stream);
-      auto const haystack_valid_it = cudf::detail::make_validity_iterator<false>(*haystack_cdv_ptr);
-
-      return thrust::find_if(
-        rmm::exec_policy(stream),
-        begin,
-        end,
-        [d_comp, haystack_valid_it] __device__(auto const idx) {
-          if (!haystack_valid_it[static_cast<size_type>(idx)]) { return false; }
-          return d_comp(idx,
-                        static_cast<rhs_index_type>(0));  // compare haystack[idx] == needle[0].
-        });
-
-    } else {
-      return thrust::find_if(
-        rmm::exec_policy(stream), begin, end, [d_comp] __device__(auto const idx) {
-          return d_comp(idx,
-                        static_cast<rhs_index_type>(0));  // compare haystack[idx] == needle[0].
-        });
-    }
-  }();
-
-  return found_it != end;
+  if (haystack.has_nulls()) {
+    auto const haystack_cdv_ptr  = column_device_view::create(haystack, stream);
+    auto const haystack_valid_it = cudf::detail::make_validity_iterator<false>(*haystack_cdv_ptr);
+
+    return thrust::any_of(
+      rmm::exec_policy(stream), begin, end, [d_comp, haystack_valid_it] __device__(auto const idx) {
+        if (!haystack_valid_it[static_cast<size_type>(idx)]) { return false; }
```

Comment:
Is it possible for the needle to be `null`? If so, what is the expected result of searching for a `null` needle in the haystack of nested data `[[1, 2], null, [3, 4]]`?

Reply:
No. That case has been addressed before reaching here, by this: https://github.com/rapidsai/cudf/blob/branch-22.08/cpp/src/search/contains.cu#L209

Reply:
Great. I would document this assumption that the needle is non-null in a code comment or as a precondition of the function.

Reply:
I've added comments in the `contains.cu` file instead, which is more suitable. This file and function don't need that: they can process null needle elements without any problem.